src/plugins/footnotes/index.ts

```diff
@@ -244,31 +276,54 @@ export default (props: PluginProps) => {
       },
 
       apply(tr, value, oldState, newState): PluginState {
+        const prevState = footnotesKey.getState(oldState)
+        if (!tr.docChanged && prevState) {
+          return prevState
+        }
         return buildPluginState(newState.doc)
       },
     },
 
     appendTransaction(transactions, oldState, newState) {
-      const { nodes: oldInlineFootnoteNodes } = footnotesKey.getState(
+      const { inlineFootnotes: oldInlineFootnoteNodes } = footnotesKey.getState(
         oldState
       ) as PluginState
 
-      const { nodes: inlineFootnoteNodes, labels } = footnotesKey.getState(
-        newState
-      ) as PluginState
+      const {
+        inlineFootnotes: inlineFootnoteNodes,
+        footnotes,
+        labels,
+        footnoteElement,
+      } = footnotesKey.getState(newState) as PluginState
+
+      // @TODO: plugin doesnt run on first launch
 
-      if (isEqual(inlineFootnoteNodes, oldInlineFootnoteNodes)) {
+      const prevIds = oldInlineFootnoteNodes.map(([node]) => node.attrs.rids)
+      const newIds = inlineFootnoteNodes.map(([node]) => node.attrs.rids)
+
+      if (!footnoteElement || isEqual(prevIds, newIds) || newIds.length < 2) {
```

Comment:
I see you are comparing the **rids** specifically instead of the whole node. Does that make a difference?

Reply:
Yes it does. isEqual is deep comparison.